module/commands/welcome.py

```diff
@@ -23,6 +23,14 @@ def generate_welcome(new_member: User) -> str:
 
 
 def send_welcome(update: Update, _: CallbackContext) -> None:
-    for new_member in update['message']['new_chat_members']:
-        if not new_member['is_bot']:
-            update.message.reply_text(f'{generate_welcome(new_member)}')
+    if update.message.new_chat_members:
+        for new_member in update.message.new_chat_members:
```

Comment:
Il controllo dell'esistenza di elementi diventa ridondante, infatti un foreach eventuale su una lista vuota non avrebbe tempo di calcolo aggiuntivo

Reply:
Scusami ma non ho capito benissimo. La miglior scelta è non eseguire il for?

Reply:
In questo caso conviene eseguirlo a prescindere, infatti il costo del for su una lista vuota e praticamente identico al controllare se esiste effettivamente qualcosa su cui iterare ed eventualmente iterare.
In breve, invece di controllare due volte l'esistenza degli elementi, conviene lasciarlo controllare dal for stesso

Reply:
Ah ceeerto, ha perfettamente senso. Sono d'accordo